Crashes after confirming photo taken with device

@roddicks I know you're not done with the FirstLaunchActivity just yet, though working on it I'm getting a crash on my device once I take my photo and tap the checkmark to confirm it. 

Does this work on your devices? Or is it a device specific issue for me?

```
ca.binder E/AndroidRuntime: FATAL EXCEPTION: main
                                                           Process: ca.binder, PID: 32383
                                                           java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=9999, result=-1, data=Intent {  }} to activity {ca.binder/ca.binder.FirstLaunchActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'boolean android.database.Cursor.moveToFirst()' on a null object reference
                                                               at android.app.ActivityThread.deliverResults(ActivityThread.java:3699)
                                                               at android.app.ActivityThread.handleSendResult(ActivityThread.java:3742)
                                                               at android.app.ActivityThread.-wrap16(ActivityThread.java)
                                                               at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1393)
                                                               at android.os.Handler.dispatchMessage(Handler.java:102)
                                                               at android.os.Looper.loop(Looper.java:148)
                                                               at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                               at java.lang.reflect.Method.invoke(Native Method)
                                                               at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                            Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'boolean android.database.Cursor.moveToFirst()' on a null object reference
                                                               at ca.binder.BinderUtilities.findImageFileFromUri(BinderUtilities.java:30)
                                                               at ca.binder.FirstLaunchActivity.onActivityResult(FirstLaunchActivity.java:106)
                                                               at android.app.Activity.dispatchActivityResult(Activity.java:6428)
                                                               at android.app.ActivityThread.deliverResults(ActivityThread.java:3695)
                                                               at android.app.ActivityThread.handleSendResult(ActivityThread.java:3742) 
                                                               at android.app.ActivityThread.-wrap16(ActivityThread.java) 
                                                               at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1393) 
                                                               at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                               at android.os.Looper.loop(Looper.java:148) 
                                                               at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                               at java.lang.reflect.Method.invoke(Native Method) 
                                                               at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 
```

--- a/src/main/java/ca/binder/FirstLaunchActivity.java
+++ b/src/main/java/ca/binder/FirstLaunchActivity.java
@@ -22,6 +22,7 @@ import java.util.ArrayList;
 import java.util.List;
 
 import ca.binder.android.DeviceInfo;
+import ca.binder.android.InternalPhoto;
 import ca.binder.domain.Profile;
 import ca.binder.remote.Server;
 import ca.binder.remote.request.UpdateProfileRequest;
@@ -31,11 +32,9 @@ import ca.binder.remote.request.UpdateProfileRequest;
  */
 public class FirstLaunchActivity extends Activity {
 
-    final String LOG_HEADER = "MYLOG###";
-    private final int IMAGE_CAPTURE_REQUEST_CODE = 9999;
-    ImageView uploadImageView;
-    File photo;
-    private Uri photopath;
+    private final int IMAGE_CAPTURE_REQUEST_CODE = 1;
+    private ImageView uploadImageView;
+    private boolean photoTaken = false;
 
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
@@ -43,9 +42,15 @@ public class FirstLaunchActivity extends Activity {
 
         uploadImageView = (ImageView)findViewById(R.id.add_user_image);
         uploadImageView.setImageDrawable(getResources().getDrawable(R.drawable.add_user));
-        setupOnClickListenerForImage();
-
         fillYearSpinner();
+
+        uploadImageView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+                startActivityForResult(intent, IMAGE_CAPTURE_REQUEST_CODE);
+            }
+        });
     }
 
 
@@ -73,37 +78,14 @@ public class FirstLaunchActivity extends Activity {
     }
 
 
-    private void setupOnClickListenerForImage() {
-        uploadImageView.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
-
-                File file = new File(Environment.getExternalStorageDirectory(), "okokok.jpg");
-                photopath = Uri.fromFile(file);
-                Log.v(LOG_HEADER, photopath.toString());
-                if (photopath != null) {
-                    intent.putExtra(MediaStore.EXTRA_OUTPUT, photopath);
-                }
-                startActivityForResult(intent, IMAGE_CAPTURE_REQUEST_CODE);
-            }
-        });
-    }
-
-
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if(requestCode == IMAGE_CAPTURE_REQUEST_CODE) {
-            if(resultCode == RESULT_OK && data != null) {
-                photo = BinderUtilities.findImageFileFromUri(photopath, this);
+        if (requestCode == IMAGE_CAPTURE_REQUEST_CODE && resultCode == RESULT_OK) {
+            Bundle extras = data.getExtras();
+            InternalPhoto photo = new InternalPhoto((Bitmap) extras.get("data"));
+            uploadImageView.setImageDrawable(photo.getDrawable(this));
 
-                Bitmap bitmap = BinderUtilities.convertFileToBitmap(photo);
-                uploadImageView.setImageBitmap(bitmap);
-                //TODO: Turn photo into image that uploadImageView can use
-            }
-            else {
-                Toast.makeText(this, "Photo was not taken", Toast.LENGTH_SHORT);
-            }
+            photoTaken = true;
         }
     }
 
@@ -224,11 +206,6 @@ public class FirstLaunchActivity extends Activity {
         returnIntent.putExtra("finished", true);
         setResult(Activity.RESULT_OK, returnIntent);
         finish();
-    }
-
-    //Ensure back does not navigate away from profile creation page
-    @Override
-    public void onBackPressed() {
     }
 
     /**

--- a/src/main/java/ca/binder/android/InternalPhoto.java
+++ b/src/main/java/ca/binder/android/InternalPhoto.java
@@ -19,17 +19,14 @@ import ca.binder.domain.IPhoto;
  */
 public class InternalPhoto implements IPhoto {
 
-    private final File file;
-    private final Context context;
+    private final Bitmap bitmap;
 
-    public InternalPhoto(File file, Context context) {
-        this.file = file;
-        this.context = context;
+    public InternalPhoto(Bitmap bitmap) {
+        this.bitmap = bitmap;
     }
 
     @Override
     public Drawable getDrawable(Context context) {
-        Bitmap bitmap = BitmapFactory.decodeFile(file.getAbsolutePath());
-        return new BitmapDrawable(this.context.getResources(), bitmap);
+        return new BitmapDrawable(context.getResources(), bitmap);
     }
 }

